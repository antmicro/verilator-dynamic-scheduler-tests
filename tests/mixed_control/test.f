$TEST_DIR/test.sv $REPO_DIR/src/main_old.cpp
