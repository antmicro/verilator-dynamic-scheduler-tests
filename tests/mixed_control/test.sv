module t(/*AUTOARG*/
   // Inputs
   clk
   );
   input clk;

   int cyc = 0;

   logic sig1 = 0;
   logic sig2 = 1;
   event ev;

   always @(posedge clk) begin
      cyc <= cyc + 1;
      if (cyc == 2) ->ev;
      else if (cyc == 4) sig1 = 1;
      else if (cyc == 6) sig2 = 0;
      else if (cyc == 7) sig1 = 0;
      else if (cyc == 8) sig1 = 1;
      else if (cyc == 10) begin
         $write("*-* All Finished *-*\n");
         $finish;
      end
   end

   initial begin
      @(posedge sig1, ev)
      $write("[%0t] Got posedge sig1 or event\n", $time);
      @(posedge sig1, ev)
      $write("[%0t] Got posedge sig1 or event\n", $time);
      @(posedge sig1, negedge sig2)
      $write("[%0t] Got posedge sig1 or negedge sig2\n", $time);
      @(posedge sig1, negedge sig2)
      $write("[%0t] Got posedge sig1 or negedge sig2\n", $time);
   end

endmodule
