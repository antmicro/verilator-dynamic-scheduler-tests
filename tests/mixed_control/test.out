[5] Got posedge sig1 or event
[9] Got posedge sig1 or event
[13] Got posedge sig1 or negedge sig2
[17] Got posedge sig1 or negedge sig2
*-* All Finished *-*
